### DISCUSSION TITLE
getting Option 'sourceMap' cannot be specified with option 'inlineSourceMap'.

### DISCUSSION THREAD

**Discussion by ShailaPatilacmeconnect** [cypress-io/cypress#29187] [Questions and Help]
even after having only "sourceMap": true    and not inlineSourceMap in tsconfig.json getting Option 'sourceMap' cannot be specified with option 'inlineSourceMap'.   have checked all other files has "sourceMap": true,

**Comment by portyu9** [comment]
Hi @ShailaPatilacmeconnect 

`"sourceMap": true` by itself is valid and is actually the setting Cypress currently recommends for TypeScript.

That error means the **effective TypeScript configuration** has both `sourceMap` and `inlineSourceMap` enabled, even if `inlineSourceMap` is not directly present in the `tsconfig.json` you are looking at.

First check the fully resolved configuration, including anything inherited through `extends`:

```bash
npx tsc --showConfig -p tsconfig.json
```

Also run it against any Cypress-specific `tsconfig.json` your project uses.

Look for:

```json
{
  "sourceMap": true,
  "inlineSourceMap": true
}
```

If `inlineSourceMap` does not appear there, then a custom Cypress preprocessor/bundler is likely injecting it. In that case, update or reconfigure that preprocessor rather than removing `"sourceMap": true`.

For Cypress's current recommendation, keep:

```json
{
  "compilerOptions": {
    "sourceMap": true
  }
}
```

and do not enable TypeScript's `inlineSourceMap`.

Please mark this answer as accepted if it helped, thank you.
